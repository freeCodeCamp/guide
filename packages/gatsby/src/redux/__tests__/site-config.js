import reducer from "../reducers/config"

describe(`add site config`, () => {
  it(`allow you to add basic site config`, () => {
    const config = {
      siteMetadata: {
        title: `yo testing`,
      },
      plugins: [],
    }
    const state = reducer(
      {},
      {
        type: `SET_SITE_CONFIG`,
        payload: config,
      }
    )
    expect(state).toMatchSnapshot()
  })

  it(`Validates configs with unsupported options`, () => {
    const config = {
      someRandomThing: `hi people`,
      plugins: [],
    }
    function runReducer() {
      return reducer(
        {},
        {
          type: `SET_SITE_CONFIG`,
          payload: config,
        }
      )
    }
    expect(runReducer).toThrowErrorMatchingSnapshot()
  })
})
