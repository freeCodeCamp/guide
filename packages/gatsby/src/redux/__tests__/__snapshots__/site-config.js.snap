// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add site config Validates configs with unsupported options 1`] = `"config is not defined"`;

exports[`add site config allow you to add basic site config 1`] = `
Object {
  "plugins": Array [],
  "siteMetadata": Object {
    "title": "yo testing",
  },
}
`;
